Civi/Api4/Generic/BasicEntity.php

```diff
@@ -132,8 +132,17 @@ public static function delete($checkPermissions = TRUE) {
    * @return BasicReplaceAction
    */
   public static function replace($checkPermissions = TRUE) {
-    return (new BasicReplaceAction(static::getEntityName(), __FUNCTION__))
+    return (new BasicReplaceAction(static::getEntityName(), __FUNCTION__, static::$idField))
```

Comment:
This was a bug. Tests caught it when I changed the name of the `id_field` for this mock entity.